readhomer_atlas/library/urn.py

```diff
@@ -0,0 +1,105 @@
+class URN:
+    """
+    Provides a subset of functionality from `MyCapytain.common.reference.URN`
+    """
+
+    NAMESPACE = 0
+    TEXTGROUP = 1
+    WORK = 2
+    VERSION = 3
+    EXEMPLAR = 4
+    NO_PASSAGE = 5
+    WORK_COMPONENT_LABELS = {
+        TEXTGROUP: "textgroup",
+        WORK: "work",
+        VERSION: "version",
+        EXEMPLAR: "exemplar",
+    }
+
+    def __str__(self):
+        return self.urn
+
+    def __init__(self, urn):
+        self.urn = urn
+        self.parsed = self.parse_urn(urn)
+
+    def parse_urn(self, urn):
+        parsed = {}
+        for v in self.WORK_COMPONENT_LABELS.values():
+            parsed[v] = None
+
+        components = urn.split(":")
+        try:
+            (
+                nid,
+                protocol,
+                namespace_component,
+                work_component,
+                passage_component,
+            ) = components[:5]
+        except ValueError:
+            raise ValueError("Invalid URN")
+        work_components = work_component.split(".")
+        parsed.update(
+            {
+                "nid": nid,
+                "protocol": protocol,
+                "namespace": namespace_component,
+                "ref": passage_component,
+            }
+        )
+        for constant, value in enumerate(work_components, 1):
+            key = self.WORK_COMPONENT_LABELS[constant]
+            parsed[key] = value
+        return parsed
+
+    @property
+    def to_namespace(self):
+        return ":".join(
+            [self.parsed["nid"], self.parsed["protocol"], self.parsed["namespace"]]
+        )
+
+    @property
+    def to_textgroup(self):
+        return ":".join([self.to_namespace, self.parsed["textgroup"]])
+
+    @property
+    def to_work(self):
+        return ".".join([self.to_textgroup, self.parsed["work"]])
+
+    @property
+    def to_version(self):
+        return ".".join([self.to_work, self.parsed["version"]])
+
+    @property
+    def to_exemplar(self):
+        return ".".join([self.to_version, self.parsed["exemplar"]])
+
+    @property
+    def to_no_passage(self):
+        if self.parsed["ref"]:
+            return self.urn.rsplit(":", maxsplit=1)[0]
+        return self.urn
+
+    def upTo(self, key):
```

Comment:
We will keep camelCase `upTo` here?

Reply:
I was aiming for parity with the URN class from [Capitains/MyCapytain](https://github.com/Capitains/MyCapytain), but even if we end up leveraging more of its functionality within ATLAS, we can and will wrap that with `up_to`.

(relates to your other questions about camel / snake below)